src/crimson/osd/main.cc

```diff
@@ -21,33 +21,66 @@ void usage(const char* prog) {
   generic_server_usage();
 }
 
+auto partition_args(seastar::app_template& app, int argc, char* argv[])
```

Comment:
If it's possible to take `argv` as `const char* []`, I would vote for it.

Reply:
it tries to align with the definition of `main()`. per n4659, § 6.6.1, the `main` function is either 

- a function of `()` or
- a function of `(int, pointer to pointer to char)`.

so, normally we have `main(int argc, char** argv)` or `main(int argc, char* argv[])`. but i cannot find a document claiming that we can (or cannot define) `argv` as a "pointer to const char*".

GCC does not complain if i define main like
```c++
main(int, const char* argv[])
```

but i am inclined to stick with the more popular signature of `(int argc, char* argv[])` before i find more supporting material.

Reply:
Sure, I'm entirely fine with that. Approving.